### ISSUE TITLE
Cannot add property _maybeTransientMute, object is not extensible

### ISSUE THREAD

**Issue by troutowicz** [daily-co/daily-js#226]
# Expected behavior

Starting a screen share of a tab with audio enabled should not crash.

# Describe the bug (unexpected behavior)

Have `user A` and `user B` in a call. Start a screen share of a tab with audio enabled as either user. A console error will print:

```
Uncaught TypeError: Cannot add property _maybeTransientMute, object is not extensible
    at Object.h [as onmute] (eval at <anonymous> (daily-iframe-esm.js:1:1), <anonymous>:30:46493)
    at i.onmute (eval at <anonymous> (daily-iframe-esm.js:1:1), <anonymous>:30:255204)
```

# Steps to reproduce

See above

# Screenshots

See above

# System information 

* Device: MacBook Air M2
* OS, version: Sonoma
* Browser, version: Arc 1.7.0

# Additional context

This crash only occurs on `@daily-co/daily-js 0.50.0`. I am not certain if its related, but we currently do not render an audio element for screen shares that have audio enabled. We only render the video element.

**Comment by mattieruth** [comment]
Thank you for filing this! We have only been able to reproduce this in our debugging environments. So I have a couple questions about your setup when you see this.
1. Do you see this when running on a secure site (https url)?
2. Are you using daily prebuilt or call object?
3. What are you using for packaging and serving?
4. Do you use babel?
5. Do you experience it on browsers other than Arc?

**Comment by troutowicz** [comment]
Hey there @mattieruth.

1. never made it to production after seeing the error locally. `http`
2. call object
3. webpack v5.75.0, webpack-dev-middleware, express
4. yes

.babelrc
```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "useBuiltIns": "entry",
        "corejs": 3
      }
    ],
    ["@babel/preset-react", {"runtime": "automatic"}]
  ],
}
```
.browserslistrc
```
>0.2%
not dead
not op_mini all
```
5. same issue on Chrome

**Comment by mattieruth** [comment]
@troutowicz -- Thanks again for reporting this. I have a fix which unfortunately just missed making it into today's release of 0.51.0. In the meantime, you might try a production build to see if the error goes away. I have not been able to track down the true root cause but it does seem only be an issue in certain dev environments.

**Comment by johnwils** [comment]
I'm also experiencing this starting in `@daily-co/daily-js 0.50.0`. If there is a fix in the works this may not be relevant, but I can reproduce the error when:

- a call connects between 2 users just after the remote `track-started` and `participant-updated` events (error shows on both ends)
- using local dev and `https` (tested using ngrok)
- using call object
- with meteor.js build system

```
// our app logs
useDailyEvents.tsx:859 track-started (remote)
channelStrip.ts:439 setSinkId
useDailyEvents.tsx:859 participant-updated (remote)
channelStrip.ts:457 audioElement:setSinkId successful tb 
channelStrip.ts:462 audioElement:play successful tb

// daily logs
call-machine-object-bundle.js:28 Uncaught (in promise) TypeError: Cannot delete property '_maybeTransientMute' of #<MediaStreamTrack>
    at Object.<anonymous> (call-machine-object-bundle.js:28:47146)
    at Generator.next (<anonymous>)
    at r (call-machine-object-bundle.js:1:2005)
    at s (call-machine-object-bundle.js:1:2215)
    at call-machine-object-bundle.js:1:2274
    at new Promise (<anonymous>)
    at Object.<anonymous> (call-machine-object-bundle.js:1:2156)
    at Object.m (call-machine-object-bundle.js:28:47237)
    at Object.v [as onunmute] (call-machine-object-bundle.js:28:46995)
    at MediaStreamTrack.<anonymous> (call-machine-object-bundle.js:28:124955)

// our app logs
useDailyEvents.tsx:859 track-started (remote)
useDailyEvents.tsx:859 participant-updated (remote)

// daily logs
call-machine-object-bundle.js:28 Uncaught TypeError: Cannot add property _maybeTransientMute, object is not extensible
    at Object.h [as onmute] (call-machine-object-bundle.js:28:46868)
    at Y.onmute (call-machine-object-bundle.js:28:124750)
```

## System Information
- Device: MacBook Pro M1
OS, version: Ventura
Browser, version: Chrome 116 and Edge 117

**Comment by wcarss** [comment]
This issue should be fixed in the just-released [daily-js 0.51.1](https://github.com/daily-co/daily-js/releases/tag/daily-js-2023-09-20-0.51.1) and will also be fixed in the upcoming 0.52.0 -- if you're on 0.50.0, unfortunately you'll have to bump forward to at least 0.51.1. Thanks for your patience, folks!
